### PR TITLE
DLS updates and general improvements (v3)

### PR DESCRIPTION
**Pull request by cshawaus**
These changes are in conjunction to updates in the https://github.com/aem-design/aemdesign-aem-support/pull/21 project.

**What they cover:**
- Renamed **Google Map** references to **Google Maps**
- Removed the analytics modules as they provided no benefit
- Improved how descriptions are allowed to appear in some components
- Fixed the order of clientlibs for the styleguide template
- Created a new DLS template
- Made several improvements to the Vue component model
- Implemented an accessor class for Sling Settings access

**Additional changes:**
- Clean up all of the models to reduce duplication
- Add new functionality and authoring capability to components such as Content Block, [Page, News, etc.] Details
- Added ability to hide authoring parsys' to prevent spacing/whitespace issues that CSS cannot resolve but requires janky JavaScript
- Component Info template replaced by toolbar action in Touch UI

### CODE REVIEW

**Comment by cshawaus** [comment]
This PR is now ready for review to be considered for develop and then a Maven release.

**Comment by wildone** [comment]
@cshawaus how are we handling datalayer component now?

File: aemdesign-aem-common/pom.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <parent>
         <groupId>design.aem</groupId>
         <artifactId>aemdesign-aem-core</artifactId>
-        <version>2.0.643</version>
+        <version>2.0.668</version>
```
Comment (wildone) [inline aemdesign-aem-common/pom.xml:11]: I agree with you to bump this to version 3.

**Comment by wildone** [comment]
@cshawaus do we want to create a depreciation component for components we are removing to ensure that upgrade of components can be clean, we can have a component print a placeholder in Authoring UI as a depreciation warning but nothin in publish view?

**Comment by cshawaus** [comment]
> @cshawaus how are we handling datalayer component now? do we need to update the showcase page?

The datalayer component still exists, what I removed was the analytics support files which were never used in projects.

I haven't removed anything to ensure backwards support remains across the board. Happy to go through it on a call to clarify any misworded points in the description.

**Comment by wildone** [comment]
AMAZING!!

Love the heading!!!

image

**Comment by cshawaus** [comment]
Opened a new PR with a correct branch name, history in this PR has been migrated automatically.

https://github.com/aem-design/aemdesign-aem-core/pull/33

**Pull request closed without merging**